### PR TITLE
fix(ci): Prevent benchmark job from being generated in MQ

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Prevent benchmark job from being generated in MQ

### Motivation
This job is not supposed to be generated in MQ (issue introduced by #24139)

### Additional Notes
Thanks @paulcacheux

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1033,6 +1033,8 @@ workflow:
     allow_failure: true
 
 .on_trace_agent_changes_or_manual:
+  - <<: *if_mergequeue
+    when: never
```
Comment (CelianR) [inline .gitlab-ci.yml:1037]: It is the same right ?
```suggestion
    - !reference [.except_mergequeue]
```

**Review by CelianR — CHANGES_REQUESTED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (chouetz) [reply .gitlab-ci.yml:1037]: It's the same but I prefer "natural" yaml references when in the same file. There is no real rule about this though..

**Review by CelianR — APPROVED** [review]
Lgtm !

**Pull request merged**